doc/advanced-queries.md

```diff
@@ -44,3 +44,38 @@ Example:
   }
 }
 ```
+### Nested Filtering
+
+Cloudant connector accepts the nested property as a single property of fields joined by `.`. 
+
+Example: a `Customer` model has a nested property `address.tags.tag`, the correct filter would be:
+```js
+// Correct
+{where: {address.tags.tag: 'home'}}
+```
+not 
+```js
+// Wrong
+{where: {address: {tags.tag: 'home'}}}
+```
+or
+```js
+// Wrong
+{where: {address: {tags: {tag: 'home'}}}}
+```
+
+#### Filtering array type property
+
+When a field in a nested property is an array, for example:
+
+In `field1.field2.field3`, `field2` is an array, Cloudant requires an operator `$elemMatch` after it to make the query work: `field1.field2.$elemMatch.field3`.
+
+To make it consistent with other connectors, user don't need to add `$elemMatch` if they define the type of each nested property in the model properly. Cloudant connector detects their data type then inserts an `$elemMatch` for array property. So take the example above, `field1.field2.field3` will work.
+
+User still have the flexibility to provide a completed query that matches Cloudant's criteria: `field1.field2.$elemMatch.field3`, in case any of nested property's type is not detectable. The connector will send that original query to the database. For details, refer to [Cloudant Query Combination Operator](https://docs.cloudant.com/cloudant_query.html#combination-operators)
```

Comment:
@0candy 
> User still have the flexibility to provide a completed query that matches Cloudant's criteria: `field1.field2.$elemMatch.field3`, in case any of the nested property's type is not detectable.

is one sentence, maybe 'in case any of..., user still have....' would be clearer?